src/worker/ingestion/hooks.ts

```diff
@@ -0,0 +1,225 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+import { captureException } from '@sentry/node'
+import { StatsD } from 'hot-shots'
+import fetch from 'node-fetch'
+import { format } from 'util'
+
+import { Action, Person } from '../../types'
+import { DB } from '../../utils/db/db'
+import { stringify } from '../../utils/utils'
+import { OrganizationManager } from './organization-manager'
+import { TeamManager } from './team-manager'
+
+export enum WebhookType {
+    Slack = 'slack',
+    Discord = 'discord',
+    Teams = 'teams',
+}
+
+export function determineWebhookType(url: string): WebhookType {
+    url = url.toLowerCase()
+    if (url.includes('slack.com')) {
+        return WebhookType.Slack
+    }
+    if (url.includes('discord.com')) {
+        return WebhookType.Discord
+    }
+    return WebhookType.Teams
```

Comment:
Is this a good approach? Probably legacy so should be fine, but while this might not be harmful, I think I'd like some explicit Teams checking with an additional unknown/unspecified type, just for clarity.

I regularly use https://webhook.site/ for example

Reply:
Well, Teams is just the default behavior this way. I don't know what webhook.site accepts and I don't want to support it, so Teams format it is.

Reply:
That makes sense yeah - webhook.site is just a webhook dump. 

When I added this comment I didn't see what the type was being used for, but yeah, if it's just formatting the message/payload, that's perfectly fine